src/ILVerify/src/SimpleArrayOfTRuntimeInterfacesAlgorithm.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using Internal.IL;
+using Internal.TypeSystem;
+
+namespace ILVerify
+{
+    internal class SimpleArrayOfTRuntimeInterfacesAlgorithm : RuntimeInterfacesAlgorithm
+    {
+        private DefType[] _arrayRuntimeInterfaces;
+        private MetadataType[] _genericRuntimeInterfaces;
+        private ModuleDesc _systemModule;
+
+        public SimpleArrayOfTRuntimeInterfacesAlgorithm(ModuleDesc systemModule)
+        {
+            _systemModule = systemModule;
+            _arrayRuntimeInterfaces = _systemModule.GetKnownType("System", "Array").RuntimeInterfaces;
+            _genericRuntimeInterfaces = new MetadataType[]
+            {
+                _systemModule.GetKnownType("System.Collections.Generic", "IEnumerable`1"),
```

Comment:
I won't insist on this, but if instead of `GetKnownType` this calls the normal `GetType` and allows these to be null, we can ILVerify .NET assemblies before the introduction of generics, or class libraries that don't want the overhead of the generic interfaces (we in fact have [a core library](https://github.com/dotnet/corert/blob/95a87f8d1c318a15e2ef77e241420f8d9bd1ed8c/src/Runtime.Base/src/System/Array.cs#L47) that doesn't have these).